src/main/java/Duke.java

```diff
@@ -1,10 +1,92 @@
+import java.util.Arrays;
+import java.util.Scanner;
 public class Duke {
+    static String horizontalLine = "--------------------------------------------------";
+    static int thingsCounted = 0;
+    static Task[] tasks = new Task[100];
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println(horizontalLine + "\nHello! I'm Duke\n" + "What can I do for you?\n" + horizontalLine);
+        storeTextAndList();
+    }
+
+    public static void storeTextAndList() {
+        String text;
+        Scanner in = new Scanner(System.in);
+        text = in.nextLine();
+        while (!text.equals("bye")) {
+            System.out.println(horizontalLine);
+            if (text.equals("list")) {
+                int count = 0;
+                for (count = 0; count < thingsCounted; count++) {
+                    System.out.print(count + 1 + '.');
+                    System.out.println(tasks[count]);
+                }
+                System.out.println(horizontalLine);
+            } else if (text.contains("done")) {
+                int index = Integer.parseInt(text.substring(5));
+                doneTask(index);
+            } else {
+                if (text.contains("todo")) {
+                    addTodoTask(text);
+                } else if (text.contains("deadline")) {
+                    addDeadlineTask(text);
+                } else {
+                    addEventTask(text);
+                }
+                System.out.println(horizontalLine);
+                text = in.nextLine();
+            }
+            System.out.println(horizontalLine);
+            System.out.println("Bye. Hope to see you again soon");
+            System.out.println(horizontalLine);
+        }
+    }
+
+    public static void addTodoTask(String text){
+        String todoDescription;
+        todoDescription = text.substring(5);
+        Task task = new Todo(todoDescription);
+        tasks[thingsCounted] = task;
+        thingsCounted++;
+        printTask(task);
+    }
+
+    public static void addDeadlineTask(String text){
+        String deadlineDescription;
+        String deadlineByDate;
+        int getIndex;
+        getIndex = text.indexOf("/");
+        deadlineDescription = text.substring(9, getIndex - 1);
+        deadlineByDate = text.substring(getIndex + 4);
+        Task task = new Deadline(deadlineDescription, deadlineByDate);
+        tasks[thingsCounted] = task;
+        thingsCounted++;
+        printTask(task);
+    }
+
+    public static void addEventTask(String text){
+        String eventDescription;
+        String eventAtDate;
+        int getIndex;
```

Comment:
Variables should be initialized where they are declared to ensure that it is valid at any time so consider moving them to when you are gonna assign them values.